### PR TITLE
Pens require gravity to write

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This was split off of https://github.com/tgstation/tgstation/pull/65668 into its own separate PR.

You cannot write on anything when there is no gravity present. Crayons, fountain pens, and charcoal pens do not require gravity to work.

Do fountain pens work in space?  [Why yes, yes they do.](https://www.youtube.com/watch?v=XS-pi0V98-w)

## Why It's Good For The Game

Pens now respect the laws of physics. 

## Changelog

:cl:
expansion: Pens now respect the laws of physics. Pens cannot write on anything when there is no gravity present. Crayons, fountain pens, and charcoal pens do not require gravity to work.
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
[The Space Pen](https://en.wikipedia.org/wiki/Space_Pen), used in real life rather than pencils due to being less flammable.

**Comment by GuillaumePrata** [comment]
As we have a gravity generator it is safe to assume that NT will give us cheap pens.
Giving a space pen for the Curator would be quite flavorful tho.

**Comment by Wallemations** [comment]
> As we have a gravity generator it is safe to assume that NT will give us cheap pens. Giving a space pen for the Curator would be quite flavorful tho.

The space pen was sold for $6 a pen, and assuming that paper-pushing-technology has advanced in the future, I'd imagine they would only have made it cheaper over time.

**Comment by Wallemations** [comment]
> [The Space Pen](https://en.wikipedia.org/wiki/Space_Pen), used in real life rather than pencils due to being less flammable.
> 
> You're right that ballpoint pens aren't used in space, but from what I can tell in the code, at no point does it mention the pens in the game being ballpoint.

On this note, it seems I was wrong, there is a *single* mention to them being ball-point, in `/datum/bounty/item/assistant/pens`'s description. In my opinion it would be easier to just remove the words "ball point" from that and just leave it at that rather than create an entire thing people have to juggle around.

**Comment by GuillaumePrata** [comment]
NT makes us pay for job equipment like foam grenades to engineers.
I just don't think it makes sense to expect space pens being given as there is a cheaper option.

Dodging all the lore talk, I also just think it is a small but interesting issue to deal with as a player for when the gravity generator stops.

File: code/modules/paperwork/pen.dm [hunk 1]
```diff
@@ -79,9 +80,10 @@
 
 /obj/item/pen/fountain
 	name = "fountain pen"
-	desc = "It's a common fountain pen, with a faux wood body."
+	desc = "It's a common fountain pen, with a faux wood body. Rumored to work in zero gravity situations"
```
Comment (JohnFulpWillard) [inline code/modules/paperwork/pen.dm:83]:
```suggestion
	desc = "It's a common fountain pen, with a faux wood body. Rumored to work in zero gravity situations."
```
sicko

**Comment by ArcaneDefence** [comment]
Having written an entry and curated the library a fair bit I don't think we need to mechanically disincentivize it in any way.
Paper code already goes through shit falling apart commonly as it is, and don't think adding a cute inconvenience is worthwhile.

**Comment by Bawhoppen** [comment]
This is the OG space meme though Arcane. We'd be missing out without it.

How about as a compromise, we add some pencils as loot on the Derelict?

**Comment by Wallemations** [comment]
I've mulled it over and I've turned around on this one, good fluff change.
One thing I do have to say though, I don't understand how a fountain pen would not be subjected to gravity.

**Comment by timothymtorres** [comment]
Fountain pens have a capillary effect which ignores gravity.

**Comment by Wallemations** [comment]
> Fountain pens have a capillary effect which ignores gravity. See: https://www.youtube.com/watch?v=XS-pi0V98-w

oh sick, you learn something new every day

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -1102,6 +1102,29 @@
...
+
+	var/obj/item/pen/pen = writing_instrument
+	var/writing_instrument_requires_gravity
+
+	if(istype(pen))
+		writing_instrument_requires_gravity = pen.requires_gravity
+	else if(istype(writing_instrument, /obj/item/toy/crayon))
+		writing_instrument_requires_gravity = FALSE
+
+	if(!has_gravity() && writing_instrument_requires_gravity)
```
Comment (Kylerace) [inline code/modules/mob/mob.dm:1122]:
```suggestion
	if(writing_instrument_requires_gravity && !has_gravity())
```

**Pull request merged**